### ISSUE TITLE
Unbox stops updating randomly with Denon Prime 4

### ISSUE THREAD

**Issue by nidde-posten** [erikrichardlarson/unbox#50]
I'm having an issue while using Unbox on Windows 10 Pro with my Prime 4

At random times the tracks will stop updating, and it'll be stuck on the latest track.

Using Force Restarts, or restarting the application usually fixes the issue, but is there something I can do to minimize the chance of it freezing on a track?

I'm not getting any obvious error messages with the latest release, whereas there used to be a Javascript error with the previous releases.

**Comment by erikrichardlarson** [comment]
Yeah we've run into this a few times on Windows, it seems to be network related. What we're going to do is check the output of the Denon poller and kill / restart the poller, basically automating the force restarts that have been working for you and others. This should be relatively straightforward to implement so I can probably have this out by end of weekend. 

Good to hear that the JS errors are gone btw, lmk if you run into any more issues!

**Comment by djrexy** [comment]
> 
> 
> I'm having an issue while using Unbox on Windows 10 Pro with my Prime 4
> 
> At random times the tracks will stop updating, and it'll be stuck on the latest track.
> 
> Using Force Restarts, or restarting the application usually fixes the issue, but is there something I can do to minimize the chance of it freezing on a track?
> 
> I'm not getting any obvious error messages with the latest release, whereas there used to be a Javascript error with the previous releases.

the same REKORDBOX and DDJ 1000

**Comment by erikrichardlarson** [comment]
@djrexy The RB / KUVO modes are separate from the Denon mode so we're probably seeing a separate issue there. It looks like it worked well on your last set until the two hour mark? https://www.twitch.tv/videos/1089999863

**Comment by iampxyco** [comment]
I have the same problem on my Denon Prime 4.
The program stops randomly from receiving data which has the consequence of no updated track listing.

Is there any fix yet?

Keep up the good work!

**Comment by erikrichardlarson** [comment]
No fix yet, working on a larger release that will be out this month that includes this update. For the time being you can close unbox and reopen it to launch the Denon listener again, we'll have this automated for you soon though.

**Comment by djrexy** [comment]
hopefully fix for randomly stop working gonna be for Recordbox too :)
Thx for good job Bro!

**Comment by djrexy** [comment]
And please implement access for Rekordbox ver6.5.3 or maybe any script where we can change version of RB on future. Is too many updates and could be painful for you to updating UNBOX every time.

**Comment by erikrichardlarson** [comment]
Yeah an update for 6.5.3 will be coming out soon, hoping that will fix the issues you've seen with 6.5.2. And yeah my plan is to implement over the air updates so you won't need to re-download unbox entirely, just refresh the app that you've already downloaded.

**Comment by djrexy** [comment]
What a NEWS!
Like always,good job and thanks for your efforts!

**Comment by djargic** [comment]
A Windows work around I found for this issue on Denon Prime products was to have the unbox_denon_poller APP running in the background at the same time as UNBOXED. I've had zero issues with dropped network connections this method.

**Comment by gustavolanzas** [comment]
Hello! I'm running into this same issue using 2 SC5000's and a x1800 mixer on a Mojave Mac. Restarting Unbox and wiggling volume faders usually gets it going again. I'm going to try and run the denon poller @djargic suggested. Thanks for the great app!

**Comment by erikrichardlarson** [comment]
Oh awesome thanks for the tip @djargic! Longer term I'm planning on using the node package that @MarByteBeep developed to implement the StageLinQ protocol which should make the Denon mode more maintainable / reliable. Also planning to get some Denon gear so I can test this better for y'all, will keep you posted here!

**Comment by djargic** [comment]
@erikrichardlarson if you need a beta tester let me know LOL

I have a SC5000 and a Prime GO.

**Comment by erikrichardlarson** [comment]
Oh I'll absolutely take you up on that haha I'll send a beta version your way when I have it ready.

**Comment by erikrichardlarson** [comment]
@djrexy 6.5.3 is now supported, you can download here: https://github.com/erikrichardlarson/unbox/releases/tag/9.2

**Comment by alldancemusic** [comment]
Hi guys, I have implemented unbox for my Denon Dj prime 4 with Windows 10.
I start unbox as administrator and the titles come out correctly, if I leave the denon in continue list mode, it changes the titles without problems.
The problem appears when, you are mixing two tracks for 2 minutes, for example a song on deck 1, a base on deck 2, for for example 2 minutes, it stops refreshing and the titles no longer change.

If I press reload function, I get a java error I think.

I have tried everything, ports, firewall, ip, I have no more idea.

If you need to test with denon prime 4 I am ready for what you need.
![cap1](https://user-images.githubusercontent.com/96615217/147403827-9e9205be-c699-43eb-85a3-30686d5b5e81.PNG)

**Comment by alldancemusic** [comment]
I have followed a user who says I install it in C and that it worked for him.

I have checked it and neither, with administrator permissions.
![cap2](https://user-images.githubusercontent.com/96615217/147404159-5fd59ffe-a70a-4bd2-96a1-443c4d3081e6.PNG)

**Comment by MarByteBeep** [comment]
@erikrichardlarson could this be something on 'my end'?

**Comment by erikrichardlarson** [comment]
@MarByteBeep No I actually haven't started using your library yet, working on that this week though. @alldancemusic I'm going to have an update here for Denon users soon, would be cool with testing the update sometime this week?

**Comment by alldancemusic** [comment]
no problem
we can stay and help as necessary

**Comment by erikrichardlarson** [comment]
@MarByteBeep @alldancemusic @djargic @gustavolanzas Just updated the Denon mode to use @MarByteBeep's StageLinq library, I don't have any Denon gear to fully test the update, could y'all give it a try and let me know if it works for you? Here are the download links: 
- [Windows ](https://github.com/erikrichardlarson/unbox/releases/download/10.0/unbox_windows.zip)
- [Mac Intel](https://github.com/erikrichardlarson/unbox/releases/download/10.0/unbox_mac_intel.zip) 
- [Mac M1](https://github.com/erikrichardlarson/unbox/releases/download/10.0/unbox_mac_arm64.zip)

**Comment by alldancemusic** [comment]
thanks, i can do the test in windows first thing tomorrow.  I keep you informed of the test

**Comment by alldancemusic** [comment]
Unzipped the files and run as administrator.
It starts with a recordbox window. I close the window to select DENON and the program crashes and closes automatically by mistake.

**Comment by iampxyco** [comment]
I unfortunately have the same problem, program crashes shortly after starting and selecting on Denon.

**Comment by erikrichardlarson** [comment]
Good to know, going to revert to the go StageLinq library while we figure out a few things with the latest release. Will let you know when I release!

**Comment by MarByteBeep** [comment]
@erikrichardlarson I just closed two issues that have been addressed by @kkirjala

https://github.com/MarByteBeep/StageLinq/issues?q=is%3Aissue+is%3Aclosed

Perhaps this also addressed the crashes mentioned above?

It would be helpful to get some logs/feedback of those crashes to narrow down the cause.

**Comment by erikrichardlarson** [comment]
Oh awesome, will push out a new release that includes these updates. Yeah I can send logs your way if I can capture some.
